go/runtime/bundle/manager.go

```diff
@@ -212,9 +237,59 @@ func (m *Manager) run(ctx context.Context) {
 	}
 }
 
-// Add adds bundle from the given path.
-func (m *Manager) Add(path string) error {
-	manifest, err := m.explodeBundle(path)
+func dynamicManifestRewriter(labels map[string]string) ManifestRewriterFunc {
+	// When no labels are specified, do not perform any rewriting.
+	if len(labels) == 0 {
+		return nil
+	}
+
+	return func(manifest *Manifest) {
+		manifest.Name = ""
+		manifest.Version = version.Version{}
+
+		// Rewrite all components.
+		for _, comp := range manifest.Components {
+			h := tuplehash.New256(32, []byte("oasis-core/bundle: dynamic component ID"))
+			// Ensure components are isolated by origin.
+			_, _ = h.Write([]byte(labels[LabelOrigin]))
```

Comment:
Should components that don't have origin label be skipped as well? Or error out in that case?